services/user-tenant-service/src/main/java/com/sourcefuse/jarc/services/usertenantservice/controller/RoleUserTenantController.java

```diff
@@ -0,0 +1,119 @@
+package com.sourcefuse.jarc.services.usertenantservice.controller;
+
+import com.sourcefuse.jarc.core.commonutils.CommonUtils;
+import com.sourcefuse.jarc.core.dto.Count;
+import com.sourcefuse.jarc.services.usertenantservice.dto.Role;
+import com.sourcefuse.jarc.services.usertenantservice.dto.UserTenant;
+import com.sourcefuse.jarc.services.usertenantservice.repository.RoleRepository;
+import com.sourcefuse.jarc.services.usertenantservice.repository.RoleUserTenantRepository;
+import jakarta.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.BeanUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+@Slf4j
+@RequestMapping("/roles")
+@RequiredArgsConstructor
+public class RoleUserTenantController {
+
+  private final RoleRepository roleRepository;
+
+  private final RoleUserTenantRepository roleUserTRepository;
+
+  @PostMapping("{id}/user-tenants")
+  public ResponseEntity<Object> createRole(
+    @Validated @RequestBody UserTenant userTenant,
+    @PathVariable("id") UUID id
+  ) {
+    UserTenant savedUserRole;
+
+    Optional<Role> role = roleRepository.findById(id);
+    if (role.isPresent()) {
+      userTenant.getRole().setId(role.get().getId());
+
+      role.get().getUserTenants().add(userTenant);
```

Comment:
This is unnecessary, remove this line of code and check the rest of the functionality will work